pandas/core/dtypes/cast.py

```diff
@@ -375,7 +374,11 @@ def trans(x):
 
 
 def maybe_cast_result(
-    result: ArrayLike, obj: Series, numeric_only: bool = False, how: str = ""
+    result: ArrayLike,
+    dtype: DtypeObj,
+    numeric_only: bool = False,
+    how: str = "",
+    same_dtype: bool = True,
```

Comment:
i would instead call this `casting`, in a similar vein to: `np.astype` (so maybe just all 'no' | 'equiv' or similar here

Reply:
could also make an enum for this

Reply:
bool is a lot clearer than an enum, doesnt require the reader to go find the enum definition